server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -148,68 +154,63 @@ public void start()
 
       final Duration delay = config.get().getPollDuration().toStandardDuration();
       exec.scheduleWithFixedDelay(
-          new Runnable()
-          {
-            @Override
-            public void run()
-            {
-              // poll() is synchronized together with start(), stop() and isStarted() to ensure that when stop() exists,
-              // poll() won't actually run anymore after that (it could only enter the syncrhonized section and exit
-              // immediately because the localStartedOrder doesn't match the new currentStartOrder). It's needed
-              // to avoid flakiness in SQLMetadataSegmentManagerTest.
-              // See https://github.com/apache/incubator-druid/issues/6028
-              readLock.lock();
-              try {
-                if (localStartOrder == currentStartOrder) {
-                  poll();
-                }
-              }
-              catch (Exception e) {
-                log.makeAlert(e, "uncaught exception in segment manager polling thread").emit();
-
-              }
-              finally {
-                readLock.unlock();
-              }
-            }
-          },
+          createPollTaskForStartOrder(localStartOrder),
           0,
           delay.getMillis(),
           TimeUnit.MILLISECONDS
       );
     }
     finally {
-      writeLock.unlock();
+      lock.unlock();
     }
   }
 
+  private Runnable createPollTaskForStartOrder(long startOrder)
+  {
+    return () -> {
+      // poll() is synchronized together with start(), stop() and isStarted() to ensure that when stop() exits, poll()
+      // won't actually run anymore after that (it could only enter the syncrhonized section and exit immediately
+      // because the localStartedOrder doesn't match the new currentStartOrder). It's needed to avoid flakiness in
+      // SQLMetadataSegmentManagerTest. See https://github.com/apache/incubator-druid/issues/6028
+      ReentrantReadWriteLock.ReadLock lock = startStopLock.readLock();
+      lock.lock();
+      try {
+        if (startOrder == currentStartOrder) {
```

Comment:
I don't see much value in that. It's normal situation, so it should be `debug()` logging if anything. But nobody ever enables debug logging in Druid. So it will just clutter the code.

Reply:
1. Well, I doubt one can expect everything the code could possibly do. :)
2. "But nobody ever enables debug logging in Druid." Subjective, you don't know :)
3. Probably somebody doesn't use debug logging mode, because there is no debug or trace logging coded at all :) Good code usually contains a good amount of trace logging, it doesn't matter if anyone plans to use it at all.
For example, you could recall the issue https://groups.google.com/d/msg/druid-user/0cVkQ3GJbZ8/KBfFsuqoBAAJ (where you cannot get data directly from a Historical node for some datasources). Guys know it exists, but it cannot be reproduced by anyone, and the lack of good trace logging makes it impossible to debug it in production, and the problem persists.